src/packet/packet.c

```diff
@@ -0,0 +1,42 @@
+/**
+ * @author  @developStorm
+ * @date    2025-05-17
+ *
+ * Implementation of packet authentication methods.
+ */
+
+#include "packet.h"
+#include "logger.h"
+
+#include <tinycrypt/hmac.h>
+#include <tinycrypt/sha256.h>
+#include <tinycrypt/utils.h>
+
+#define PACKET_HMAC_PSK "StanfordSSIUMoJBTXpnXoZX3bQrNm5f"
+#define PACKET_HMAC_PSK_LEN 32
+
+_Static_assert(offsetof(packet_t, hmac) ==
```

Comment:
How does this `_Static_assert` work? (something pulled out from one of the tinycrypt files?

`hmac` variable here seems like it's floating undefined until line 31 (local function stack variable) unless I'm missing something here.

Reply:
Ah thanks! That's a useful check!